common/src/main/java/org/apache/celeborn/common/util/ShuffleBlockInfoUtils.java

```diff
@@ -34,17 +34,28 @@ public static List<Long> getChunkOffsetsFromShuffleBlockInfos(
       int startMapIndex,
       int endMapIndex,
       long fetchChunkSize,
-      Map<Integer, List<ShuffleBlockInfo>> indexMap) {
+      Map<Integer, List<ShuffleBlockInfo>> indexMap,
+      List<Long> sortedFileLenSumOver) {
     List<Long> sortedChunkOffset = new ArrayList<>();
     ShuffleBlockInfo lastBlock = null;
+    int sumOverIdx = 0;
     for (int i = startMapIndex; i < endMapIndex; i++) {
       List<ShuffleBlockInfo> blockInfos = indexMap.get(i);
       if (blockInfos != null) {
         for (ShuffleBlockInfo info : blockInfos) {
           if (sortedChunkOffset.size() == 0) {
             sortedChunkOffset.add(info.offset);
+            if (sortedFileLenSumOver.size() > 1) {
+              while (sortedFileLenSumOver.get(sumOverIdx + 1) < info.offset) {
```

Comment:
while (sortedFileLenSumOver.get(sumOverIdx + 1) -> while (sortedFileLenSumOver.get(sumOverIdx )?

Reply:
Oh, in my view, it is sumOverIdx+1.